### PR TITLE
feat: added add label feature

### PR DESCRIPTION
**Pull request by paarthbajaj**
label

What's added in this PR:
User can add only one label for any note added but not via checkboxes

### CODE REVIEW

**Review by kedark152 — COMMENTED** [review]
Code is written very well,
Just work on UI part once you get time.

File: src/components/ShowNote.js [hunk 1]
```diff
@@ -43,7 +47,14 @@ export const ShowNote = () => {
                   className="fal fa-edit cursor-pointer"
                   onClick={() => editNote(noteItem)}
                 />
-                <i className="fal fa-tag cursor-pointer" />
+                <i
+                  className="fal fa-tag cursor-pointer"
+                  onClick={() => {
+                    setIsTagPopupOpen(() => true);
+                    setSelectedNote(() => noteItem);
+                    // setCounter((counter) => counter + 1);
```
Comment (rkapoor10) [inline src/components/ShowNote.js:55]: remove comment if not needed

File: src/context/NotesContext.js [hunk 2]
```diff
@@ -28,13 +28,16 @@ const NotesContextProvider = ({ children }) => {
...
   const [noteContent, setNoteContent] = useState({
     title: "",
     body: "",
     in_trash: false,
     color: "white",
+    createdAt: "",
+    tags: [],
   });
   const deleteNote = async (noteItem) => {
     console.log(noteItem);
```
Comment (rkapoor10) [inline src/context/NotesContext.js:43]: remove console.log

File: src/context/NotesContext.js [hunk 3]
```diff
@@ -125,12 +119,49 @@ const NotesContextProvider = ({ children }) => {
...
         }
       );
-      console.log("updatedNote", updatedNoteRes);
     } catch (err) {
       console.log(err);
     }
   };
-
+  const editLabel = (e) => {
+    console.log({
```
Comment (rkapoor10) [inline src/context/NotesContext.js:127]: here also

File: src/context/NotesContext.js [hunk 4]
```diff
@@ -125,12 +119,49 @@ const NotesContextProvider = ({ children }) => {
...
+      tags: [e.target.value],
+    });
+    setSelectedNote(() => ({
+      ...selectedNote,
+      tags: [e.target.value],
+    }));
+  };
+  const addMultiLabel = (e) => {
+    console.log(e.target.value);
+    console.log(e.target.checked);
```
Comment (rkapoor10) [inline src/context/NotesContext.js:138]: here also

File: src/context/NotesContext.js [hunk 5]
```diff
@@ -125,12 +119,49 @@ const NotesContextProvider = ({ children }) => {
...
+    setCounter((counter) => counter + 1);
+    console.log({
+      ...selectedNote,
```
Comment (rkapoor10) [inline src/context/NotesContext.js:141]: here also

File: src/pages/HomePage.js [hunk 6]
```diff
@@ -2,7 +2,7 @@
 // import Searchbar from "../components/Searchbar";
 // import AddNote from "../components/AddNote";
 // import ShowNote from "../components/ShowNote";
```
Comment (rkapoor10) [inline src/pages/HomePage.js:4]: remove comments if not needed

**Review by rkapoor10 — APPROVED** [review]
The code looks good to merge just go through the comments to clean up code

File: src/components/TagsPopup.js [hunk 7]
```diff
@@ -0,0 +1,55 @@
...
+    addLabel,
+    notesList,
+    addMultiLabel,
+  } = useNotes();
+  let labelList = [];
+  (() => {
+    notesList?.notes?.forEach((i) => {
+      labelList = labelList.concat(i?.tags);
+    });
+  })();
```
Comment (JantuDeb) [inline src/components/TagsPopup.js:18]:
```
const labelList = notesList?.notes.map(item=>item.tags)

```

File: src/components/TagsPopup.js [hunk 8]
```diff
@@ -0,0 +1,55 @@
...
+            </form>
+            {labelList?.length > 0 &&
+              labelList.map((i) => (
+                <label key={i}>
+                  <input
+                    type="checkbox"
+                    value={i}
+                    onClick={(e) => {
+                      addMultiLabel(e);
+                    }}
```
Comment (JantuDeb) [inline src/components/TagsPopup.js:41]: ```onClick={addMultiLabel}```

**Review by JantuDeb — APPROVED** [review]
Added few suggestions

**Pull request merged**